src/ChannelServer/Util/GmCommands.cs

```diff
@@ -542,6 +543,44 @@ private CommandResult HandleRecall(ChannelConnection conn, Character sender, Cha
 			return CommandResult.Okay;
 		}
 
+		private CommandResult HandleRecallAll(ChannelConnection conn, Character sender, Character target, string command, string[] args)
+		{
+			if (args.Length > 1)
+			{
+				return CommandResult.InvalidArgument;
+			}
+
+			// TODO: Once we have support for more than one map server,
+			//   we have to search for characters across all of them.
+
+			var characters = ChannelServer.Instance.World.GetCharacters();
+
+			// If the player count exceeds the max. channel capacity, cancel command.
+			if (characters.Count() > Shared.Const.Channel.ZoneMaxPcCount)
+			{
+				sender.ServerMessage("Player count exceeds channel capacity of {0}.", Shared.Const.Channel.ZoneMaxPcCount);
+				return CommandResult.Okay;
+			}
+
+			// Recall each player to current location.
+			foreach (var character in characters)
+			{
+				if (character.Id != target.Id)
+				{
+					var location = target.GetLocation();
+					character.Warp(location);
+
+					character.ServerMessage("You've been warped to {0}.", location);
+
+					return CommandResult.Okay;
```

Comment:
By returning here, from inside the loop, you're basically canceling the command after the first successful warp, recalling only the first character.

Reply:
Thanks for the information, I could only test with 2 instances of ToS.